src/Redux.elm

```diff
@@ -24,25 +25,20 @@ view model =
 program :
     { init : ( model, Cmd msg )
     , update : msg -> model -> ( model, Cmd msg )
+    , encode : model -> Json.Value
```

Comment:
Outgoing ports are restricting to certain values (http://guide.elm-lang.org/interop/javascript.html) so I think that this is the only type that is flexible enough to cover all use cases. 

What do you mean, you would prefer dispatch would decode it?